Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -73,48 +76,176 @@ public void RoslynConfig_NoAssemblies()
         context.AssertExpectedPluginsRequested([]);
     }
 
-    [TestMethod]
-    public void RoslynConfig_ValidProfile()
+    [DataTestMethod]
+    [DataRow(RoslynAnalyzerProvider.CSharpLanguage)]
+    [DataRow(RoslynAnalyzerProvider.VBNetLanguage)]
+    public void RoslynConfig_ValidProfile_WithLegacy(string language)
+    {
+        var sonarProperties = new ListPropertiesProvider(new Dictionary<string, string>
+        {
+            {"wintellect.pluginKey", "wintellect"},
+            {"wintellect.pluginVersion", "1.13.0"},
+            {"wintellect.staticResourceName", "wintellect.zip"},
+            {"sonar.cs.analyzer.dotnet.pluginKey", "cs"},
+            {"sonar.cs.analyzer.dotnet.pluginVersion", "1.42.0"},
+            {"sonar.cs.analyzer.dotnet.staticResourceName", "SonarAnalyzer.zip"},
+            {"sonar.vbnet.analyzer.dotnet.pluginKey", "vbnet"},
+            {"sonar.vbnet.analyzer.dotnet.pluginVersion", "1.42.0"},
+            {"sonar.vbnet.analyzer.dotnet.staticResourceName", "SonarAnalyzer.zip"},
+            {"sonar.vbnet.testPropertyPattern", "foo"},
+            {"sonaranalyzer-cs.analyzerId", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-cs.ruleNamespace", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-cs.pluginKey", "cs"},
+            {"sonaranalyzer-cs.staticResourceName", "SonarAnalyzer.zip"},
+            {"sonaranalyzer-cs.nuget.packageId", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-cs.pluginVersion", "1.42.0"},
+            {"sonaranalyzer-cs.nuget.packageVersion", "1.13.0"},
+            {"sonaranalyzer-vbnet.analyzerId", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-vbnet.ruleNamespace", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-vbnet.pluginKey", "vbnet"},
+            {"sonaranalyzer-vbnet.staticResourceName", "SonarAnalyzer.zip"},
+            {"sonaranalyzer-vbnet.nuget.packageId", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-vbnet.pluginVersion", "1.42.0"},
+            {"sonaranalyzer-vbnet.nuget.packageVersion", "1.13.0"},
+            {"sonar.cs.testPropertyPattern", "foo"},
+            {"sonar.sources", "**/*.*"},
+            {"sonar.cs.foo", "bar"},
+            {"sonar.vbnet.foo", "bar"},
+            {"sonar.cs.analyzer.security.pluginKey", "securitycsharpfrontend" },
+            {"sonar.cs.analyzer.security.pluginVersion", "1.42.0" },
+            {"sonar.cs.analyzer.security.staticResourceName", "SecurityAnalyzer.zip" },
+            {"sonaranalyzer.security.cs.pluginKey", "OLDSecurityCSharpFrontend" },
+            {"sonaranalyzer.security.cs.pluginVersion", "OLDSecurityCSharpFrontend" },
+            {"sonaranalyzer.security.cs.staticResourceName", "OLDSecurityCSharpFrontend" },
+        });
+        var context = new Context(TestContext, sonarProperties, [[@"c:\assembly1.dll"], [@"d:\foo\assembly2.dll"]], null, language);
+        var securityProperties = language == RoslynAnalyzerProvider.CSharpLanguage ?
```

Comment:
`securityProperties` and `expectedSonarLintXml` are the same for `RoslynConfig_ValidProfile_WithLegacy` and `RoslynConfig_ValidProfile_WithoutLegacy`.

Extract it. And use the same scaffolding (add security) also in `RoslynConfig_ValidProfile_LegacyOnly`. Then it will be used 3x.